spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -100,7 +100,13 @@ public Filter[] pushFilters(Filter[] filters) {
     List<Filter> pushed = Lists.newArrayListWithExpectedSize(filters.length);
 
     for (Filter filter : filters) {
-      Expression expr = SparkFilters.convert(filter);
+      Expression expr = null;
+      try {
+        expr = SparkFilters.convert(filter);
+      } catch (IllegalArgumentException e) {
+        // converting to Iceberg Expression failed, so this expression cannot be pushed down
```

Comment:
We should probably log something here

Reply:
Yea there was a similar comment on : https://github.com/apache/iceberg/pull/5204#discussion_r914817448, but seemed the positive filters.  I'm ok to log unsuccessful filters too (could also add the exprs not successfully bound as well in the next statement)

Reply:
It could be useful to know that a filter was intentionally not pushed down for a particular reason

Reply:
I am OK to have a follow-up to log the unsuccessful filters and also the unsuccessfully bound exprs in the next statement